File: ESTRUTURAS/1022/main.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8')
var lines = input.split('\n')

function addition(n1, d1, n2, d2) {
    return [n1 * d2 + n2 * d1, d1 * d2]
}

function subtraction(n1, d1, n2, d2) {
    return [n1 * d2 - n2 * d1, d1 * d2]
}

function multiplication(n1, d1, n2, d2) {
    return [n1 * n2, d1 * d2]
}

function division(n1, d1, n2, d2) {
    return [n1 * d2, n2 * d1]
}

function simplify(nl, dl) {
    let smaller = Math.abs(nl) < Math.abs(dl) ? Math.abs(nl) : Math.abs(dl)
    for (let i = smaller; i > 0; i--)
        if ((nl % i === 0) && (dl % i === 0))
            return [parseInt(nl / i), parseInt(dl / i)]
}

const selector = { '+': addition, '-': subtraction, '*': multiplication, '/': division }
let N = lines[0]
for (let i = 1; i <= N; i++) {
    let [n1, , d1, op, n2, , d2] = lines[i].split(' ')
    let [nl, dl] = selector[op](n1, d1, n2, d2)
    let [nr, dr] = simplify(nl, dl)
    console.log(`${nl}/${dl} = ${nr}/${dr}`)
}

File: ESTRUTURAS/1022/main.py
def addition(N1, N2, D1, D2):
    return N1 * D2 + N2 * D1, D1 * D2


def subtraction(N1, N2, D1, D2):
    return N1 * D2 - N2 * D1, D1 * D2


def multiplication(N1, N2, D1, D2):
    return N1 * N2, D1 * D2


def division(N1, N2, D1, D2):
    return N1 * D2, N2 * D1


def simplify(NL, DL):
    smaller = abs(NL) if abs(NL) < abs(DL) else abs(DL)
    for i in range(smaller, 0, -1):
        if (NL % i == 0) and (DL % i == 0):
            return NL // i, DL // i


selector = {'+': addition, '-': subtraction, '*': multiplication, '/': division}
N = int(input())
for i in range(N):
    N1, _, D1, OP, N2, _, D2 = input().split()
    NL, DL = selector.get(OP)(int(N1), int(N2), int(D1), int(D2))
    NR, DR = simplify(NL, DL)
    print('{0}/{1} = {2}/{3}'.format(NL, DL, NR, DR))
